app.py

```diff
@@ -43,69 +54,65 @@ def profile():
     """
     The profile page of the app.
     """
+
+    if flask.request.method == "GET":
+        return render(data=current_user.profile.json)
+
     form = ProfileForm()
 
-    height_feet = form.height_feet.data
-    height_inches = form.height_inches.data
-
-    print(form.birthdate.data)
-    print(form.gender.data)
-    print(form.height_feet.data)
-    print(form.height_inches.data)
-    if height_feet is not None:
-        height = (int(height_feet) * 12) + int(height_inches)
-        print(height)
-
-    if form.validate_on_submit():
-        user_profile = Profile(
-            gender=form.gender.data,
-            height=height,
-            weight=form.weight.data,
-        )
-        db.session.add(user_profile)
-        db.session.commit()
-        print("User updated")
-        flash("Profile Updated!")
-        return redirect(url_for("profile"))
-
-    return render_template("profile.html", form=form)
-
-
-# NOTE user cannot be verified until DB is set up and conditional statements
-# will have to be updated
+    if not form.validate_on_submit():
+        return render(data=current_user.profile.json, form=form)
+
+    # delete existing profile (loop just in case)
+    for existing_profile in Profile.query.filter_by(user_id=current_user.id):
+        db.session.delete(existing_profile)
```

Comment:
What're we deleting the profile information for? 

Is this when the user wants to enter in new info?

Reply:
A user should have a single associated `Profile`, so this is ensuring that the profile is unique before updating it. This could probably be done in a cleaner way using whatever SQLAlchemy's form of `UPDATE` is.

Honestly, considering the fact that it's a one-to-one relationship, it would probably be simpler to move everything from the `Profile` table to `BTUser`. That being said, this approach works too. "If it ain't broke, don't fix it", I suppose.

Reply:
Oh ok gotcha. That's true, I think it's fine having them separated due to password safety since its stored in BTUser and not Profile.